### PR TITLE
Psmr24 pet introductory

### PR DESCRIPTION
**Pull request by NicoleJurjew**
Added solutions to a few notebooks and changed some wording

### CODE REVIEW

File: notebooks/PET/Solution_Snippets/DIY_OSEM_01_MLEM.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def MLEM(acquired_data, acq_model, initial_image, num_iterations):
+    estimated_image = initial_image.clone()
+    sensitivity = acq_model.backward(acquired_data.get_uniform_copy(1))
+    for i in range(num_iterations):
+        quotient = acquired_data/acq_model.forward(estimated_image.clone())  # y / (Ax + b)
```
Comment (KrisThielemans) [inline notebooks/PET/Solution_Snippets/DIY_OSEM_01_MLEM.py:5]:
```suggestion
        quotient = acquired_data/acq_model.forward(estimated_image)  # y / (Ax + b)
```

File: notebooks/PET/Solution_Snippets/DIY_OSEM_01_MLEM.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+        quotient.fill(numpy.nan_to_num(quotient.as_array()))
+        mult_update = acq_model.backward(quotient.clone())/sensitivity         # A^t * quotient / A^t1
```
Comment (KrisThielemans) [inline notebooks/PET/Solution_Snippets/DIY_OSEM_01_MLEM.py:7]:
```suggestion
        mult_update = acq_model.backward(quotient)/sensitivity         # A^t * quotient / A^t1
```

File: notebooks/PET/Solution_Snippets/DIY_OSEM_02_OSEM.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+def OSEM(acquired_data, acq_model, initial_image, num_iterations):
+    estimated_image=initial_image.clone()
+    # some stuff here - hint, this will be similar to your solution for MLEM
+    # but you will have to additionally iterate over your subsets 
+    for i in range(num_iterations):
+        for s in range(acq_model.num_subsets):
+
+            subs_sensitivity = acq_model.backward(acquired_data.get_uniform_copy(1), subset_num=s)
+
+            quotient = acquired_data/acq_model.forward(estimated_image.clone(), subset_num=s)     # y / (Ax + b)
```
Comment (KrisThielemans) [inline notebooks/PET/Solution_Snippets/DIY_OSEM_02_OSEM.py:10]:
```suggestion
            quotient = acquired_data/acq_model.forward(estimated_image, subset_num=s)     # y / (Ax + b)
```

File: notebooks/PET/Solution_Snippets/DIY_OSEM_02_OSEM.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            quotient.fill(numpy.nan_to_num(quotient.as_array()))
+                
+            mult_update = acq_model.backward(quotient.clone(), subset_num=s)/subs_sensitivity     # A^t * quotient / A^t1
```
Comment (KrisThielemans) [inline notebooks/PET/Solution_Snippets/DIY_OSEM_02_OSEM.py:13]:
```suggestion
            mult_update = acq_model.backward(quotient, subset_num=s)/subs_sensitivity     # A^t * quotient / A^t1
```

File: notebooks/PET/Solution_Snippets/DIY_OSEM_02_OSEM.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            mult_update.fill(numpy.nan_to_num(mult_update.as_array()))
+            
+            estimated_image *= mult_update                                                        # update (in place)
+
+            est_img_arr = estimated_image.as_array()
+            est_img_arr[est_img_arr<0] = 0
```
Comment (KrisThielemans) [inline notebooks/PET/Solution_Snippets/DIY_OSEM_02_OSEM.py:19]: we have `estimated_image.maximum(0)` now

File: notebooks/PET/Solution_Snippets/OSEM_reconstruction_01_create_acq_model.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+def create_acq_model(attn_image, background_term):
+    '''create a PET acquisition model.
+    
+    Arguments:
+    attn_image: the mu-map
+    background_term: bakcground-term as an sirf.STIR.AcquisitionData
```
Comment (KrisThielemans) [inline notebooks/PET/Solution_Snippets/OSEM_reconstruction_01_create_acq_model.py:6]:
```suggestion
    background_term: background-term as a sirf.STIR.AcquisitionData
```

File: notebooks/PET/Solution_Snippets/OSEM_reconstruction_03_create_acq_model_noAtt.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+def create_acq_model(attn_image, background_term, AC = False):
+    '''create a PET acquisition model.
+    
+    Arguments:
+    attn_image: the mu-map
+    background_term: bakcground-term as an sirf.STIR.AcquisitionData
```
Comment (KrisThielemans) [inline notebooks/PET/Solution_Snippets/OSEM_reconstruction_03_create_acq_model_noAtt.py:6]:
```suggestion
    background_term: background-term as a sirf.STIR.AcquisitionData
```

File: notebooks/PET/image_creation_and_simulation.ipynb [hunk 8]
```diff
@@ -225,7 +227,7 @@
     "with $\\mu(x)$ the linear attenuation coefficients (roughly proportional to density), \n",
     "and the line integral being performed between the 2 detectors.\n",
     "\n",
-    "It is a specific \"feature\" of PET that the attenuation factor only depends on the line integral between the detectors, and not on the voxel location (this is different in SPECT). Therefore, it is common practice to model PET acquisitions by first doing ray-tracing, followed by multiplication by attenuation factors (which have the \"shape\" of acquired data)."
+    "It is a specific \"feature\" of PET that the attenuation factor only depends on the line integral between the detectors, and not on the voxel location (this is different in SPECT). Therefore, it is common practice to model PET acquisitions by first doing ray-tracing (which is essentially a line integral), followed by multiplication by attenuation factors (which have the \"shape\" of acquired data). In SIRF, the Acquisition Sensitivity model takes care of the exponential for us!"
```
Comment (KrisThielemans) [inline notebooks/PET/image_creation_and_simulation.ipynb:230]:
```suggestion
    "It is a specific \"feature\" of PET that the attenuation factor only depends on the line integral between the detectors, and not on the voxel location (this is different in SPECT). Therefore, it is common practice to model PET acquisitions by first doing ray-tracing (which is essentially a line integral), followed by multiplication by attenuation factors (which have the \"shape\" of acquired data). In SIRF, the AcquisitionSensitivityModel takes care of the line integral over the mu-map and the exponential for us!"
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
great! Some small typos, and a few `clone()` which shouldn't be necessary. Try to "batch" suggestions that you want to accept (which can be done from the Files tab i think).

Are you sure you ran `nbstripout` on this?

**Comment by KrisThielemans** [comment]
@NicoleJurjew don't forget to add your name (and others) to the notebooks where you make large changes, or provide the solutions.

**Review by KrisThielemans — APPROVED** [review]
let me know when ready

**Comment by NicoleJurjew** [comment]
Hey Kris, this should be ready to go now! I definitely ran nbstripout, I believe creating a "new cell" still look awful when committed. There seems to be a bug though, as VSCode removes and then immediately re-adds files right after running nbstripout, so sorry. Let me know if I should add anything else. Thanks!

**Comment by KrisThielemans** [comment]
I'll merge this. Please create a new PR with extra authors if appropriate.

**Pull request merged**